server/views.py

```diff
@@ -182,6 +183,72 @@ def card(request, tool_id, card_id):
 
   return makeResponse(request, result)
 
+@check_secret
+@check_ip
+@require_GET
+def getstockinfo(request, tool_id):
+  try:
+    m = MachineItem.objects.filter(tool__pk=tool_id)
+  except ObjectDoesNotExist as e:
+        result = {'status' : 'Error'}
+        return HttpResponse(json.dumps(result), content_type='application/json')
+  result = {'status' : 'Success'}
+  result['items'] = []
+  for item in m:
+    result['items'].append({'position' : item.position, 'name' : item.item.name, 'stock' : item.stock , 'price' : item.item.price, 'position' : item.position})
+  return HttpResponse(json.dumps(result), content_type='application/json')
+
+
+@check_secret
+@check_ip
+@require_GET
+def updatebalance(request, tool_id, new_balance, card_id):
```

Comment:
For a vending machine, isn't a more useful API "deduct from balance" - retrieving then writing back is a fairly classic race condition that should be avoided with money